Performance improvements

bclyons12 opened the pull request:
I couldn't speed up the eigensolver much, but these changes make a single run of ActorTJLF go from
`1.157755 seconds (7.23 M allocations: 1.870 GiB, 19.66% gc time)`
to
`1.013869 seconds (2.14 M allocations: 1.463 GiB, 10.13% gc time)`

avdeevag commented:
I fix number of regression tests and added a couple more in this PR #47, if you can probably merge that branch into yours? Or at least fix number of tests here (https://github.com/ProjectTorreyPines/TJLF.jl/blob/master/test/runtests_regressions.jl) (only tglf03 should be excluded)?

bclyons12 commented:
#47 was merged in here, so we picked up these changes too:
- Removed dependency on Arpack package (annoying to keep the right version; probably package is not maintained anymore; not thread save) and replaced it with KrylovKit eigensolver
-  added regression test for the case where EIGEN_SPECTRUM is provided in the input file (actually this is a bad example, as both Arpack and KrylovKit eigensovers failed on this case and gesv+geev has been used instead)
-  add lines to run several regression tests with KrylovKit eigensolver instead of gesv. I noticed that even with slightly different eigenvectors (tol ~1e-3) regression test pasted.
-  added the line to use one thread for BLAS library (FUSE already has this line)
